src/MediaWiki/Hooks/SkinTemplateNavigation.php

```diff
@@ -45,6 +45,7 @@ public function __construct( SkinTemplate &$skinTemplate, array &$links ) {
 	public function process() {
 
 		if ( $this->skinTemplate->getUser()->isAllowed( 'purge' ) ) {
+			$skin->getOutput()->addModules( 'ext.smw.purge' );
```

Comment:
Here we come to a more delicate issue on the reference of `$skin->getOutput()`.

Lets take the `$skin` variable, PHP is expected to provide a reference to the variable with the name `$skin` but when you look in the code then you cannot find any reference or declaration with that name. We should select `$this->skinTemplate` because this is a variable available to us and doing `$this->skinTemplate->getOutput()...` should just do what we want.